Highway.h

```diff
@@ -13,4 +20,5 @@ struct Highway
     void removeVehicleInternal(Vehicle* v);
     int speedLimit = 65;
     std::vector<Vehicle*> vehicles;
```

Comment:
You can forward-declare `Vehicle` since you don't use it in this header file at all.  it's only used in the cpp file.
- [ ] Don't include what you don't use.